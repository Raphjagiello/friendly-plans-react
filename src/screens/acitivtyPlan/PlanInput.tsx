import React from 'react';
import { StyleSheet, View } from 'react-native';

import { Icon, TextInput } from '../../components';

export class PlanInput extends React.PureComponent {
  render() {
    return (
      <View style={styles.container}>
        <Icon name="emoticon-happy-outline" size={24} color="#C8CBFA" />
        <TextInput placeholder="Wpisz nazwę planu" />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    maxWidth: '40%',
  },
});
