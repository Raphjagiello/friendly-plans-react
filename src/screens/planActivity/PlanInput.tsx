import React, { SFC } from 'react';
import { StyleSheet, TextInputProps } from 'react-native';

import { i18n } from 'locale';
import { palette } from 'styles';
import { StyledText, TextInput } from '../../components';

interface Props extends TextInputProps {
  value: string;
  disabled?: boolean;
}

export const PlanInput: SFC<Props> = ({ value, disabled, ...props }) => (
  <>
    {disabled ? (
      <StyledText style={styles.styledText}>{value}</StyledText>
    ) : (
      <TextInput
        style={styles.textInput}
        placeholder={i18n.t('planActivity:planNamePlaceholder')}
        value={value}
        {...props}
      />
    )}
  </>
);

const styles = StyleSheet.create({
  textInput: {
    marginLeft: 8,
  },
  styledText: {
    color: palette.textBlack,
    marginLeft: 8,
  },
});
