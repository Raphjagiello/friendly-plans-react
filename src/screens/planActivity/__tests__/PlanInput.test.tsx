import { shallow } from 'enzyme';
import React from 'react';
import { PlanInput } from '../PlanInput';

describe('PlanInput', () => {
  it('renders correctly', () => {
    const planInput = shallow(<PlanInput value="Music" />);

    expect(planInput).toMatchSnapshot();
  });
});
