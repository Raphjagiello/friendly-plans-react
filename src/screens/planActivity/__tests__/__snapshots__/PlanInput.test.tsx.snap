// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanInput renders correctly 1`] = `
<Fragment>
  <TextInput
    placeholder="Type plan name"
    style={
      Object {
        "marginLeft": 8,
      }
    }
    value="Music"
  />
</Fragment>
`;
